projects/packages/global-styles-webfonts-introspector/src/class-global-styles-webfonts-introspector.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Global styles webfonts introspector
+ *
+ * @package automattic/global-styles-webfonts-introspector
+ * @since 0.1.0
+ */
+
+namespace Automattic\Jetpack\Fonts;
+
+/**
+ * Global styles webfonts introspector.
+ */
+class Global_Styles_Webfonts_Introspector {
+	/**
+	 * Extract the font family slug from a settings object.
+	 *
+	 * @param object $setting The setting object.
+	 *
+	 * @return string|void
+	 */
+	private static function extract_font_slug_from_setting( $setting ) {
+		if ( isset( $setting['typography'] ) && isset( $setting['typography']['fontFamily'] ) ) {
+			$font_family = $setting['typography']['fontFamily'];
+
+			// Full string: var(--wp--preset--font-family--slug).
+			// We do not care about the origin of the font, only its slug.
+			preg_match( '/font-family--(?P<slug>.+)\)$/', $font_family, $matches );
+
+			if ( isset( $matches['slug'] ) ) {
+				return $matches['slug'];
+			}
+
+			// Full string: var:preset|font-family|slug
+			// We do not care about the origin of the font, only its slug.
+			preg_match( '/font-family\|(?P<slug>.+)$/', $font_family, $matches );
+
+			if ( isset( $matches['slug'] ) ) {
+				return $matches['slug'];
+			}
+
+			return $font_family;
```

Comment:
Could you describe what this line would return? Would it just be a font family css variable `var(--wp--preset--font-family--slug)` where a match isn't found?

Reply:
>Also completely unrelated to this PR, but it seems so odd to have so many different ways of representing the font family slug 🙃

Right! That's because you can specify a CSS variable to link it inside `theme.json`. You can also pick a font in the editor, and then it would become the second format `var:preset|font-family|slug`, which is the one that's registered in the global styles CPT.

But then, you can also solely specify the slug, and that's why we're returning... If none of the formats match, just return what's in there and we'll try our luck. It might be a system font, might be just the slug... Who knows.